tools/serverpod_cli/test/analyzer/entities/stateful_analyzer/protocol_validation/enum_serialized_as_test.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:serverpod_cli/src/analyzer/entities/definitions.dart';
+import 'package:serverpod_cli/src/analyzer/entities/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/protocol_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test(
+      'Given a valid enum definition when validating then serializeAs is set to int.',
+      () {
+    var protocols = [
+      ProtocolSourceBuilder().withYaml(
+        '''
+        enum: ExampleEnum
+        values:
+          - first
+          - second
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(protocols, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as EnumDefinition;
+
+    expect(definition.serializeAs, SerializeEnumAs.int);
+  });
+
+  test(
+      'Given a valid enum definition with serializeAs set to int when validating then no errors are collected',
+      () {
+    var protocols = [
+      ProtocolSourceBuilder().withYaml(
+        '''
+        enum: ExampleEnum
+        serializeAs: int
+        values:
+          - first
+          - second
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(protocols, onErrorsCollector(collector));
+
+    analyzer.validateAll();
+
+    expect(collector.errors, isEmpty);
+  });
+
+  test(
+      'Given a valid enum definition with serializeAs set to int when validating then serializeAs is set to int.',
+      () {
+    var protocols = [
+      ProtocolSourceBuilder().withYaml(
+        '''
+        enum: ExampleEnum
+        serializeAs: int
+        values:
+          - first
+          - second
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(protocols, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as EnumDefinition;
+
+    expect(definition.serializeAs, SerializeEnumAs.int);
+  });
+
+  test(
+      'Given a valid enum definition with serializeAs set to string when validating then serializeAs is set to string.',
+      () {
+    var protocols = [
+      ProtocolSourceBuilder().withYaml(
+        '''
+        enum: ExampleEnum
+        serializeAs: String
+        values:
+          - first
+          - second
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    var analyzer = StatefulAnalyzer(protocols, onErrorsCollector(collector));
+
+    var definitions = analyzer.validateAll();
+
+    var definition = definitions.first as EnumDefinition;
+
+    expect(definition.serializeAs, SerializeEnumAs.string);
+  });
```

Comment:
Suggestion (Non blocking): Group the setup for these tests together in the same group and have two tests in the group.

I think we have used this pattern in the other test files to reduce repetition.